### DISCUSSION TITLE
Warning regarding CommonJS or AMD dependencies when building app

### DISCUSSION THREAD

**Discussion by ghost** [ngneat/transloco#694] [Q&A] [answered]
Hi,

Just evaluating transloco as an alternative ngx-translate as it seems to be a more buoyant community however upon installation we get the following warning: -

Warning: C:\Caboodal\Auxilio\src\Caboodal.Auxilio.Client\node_modules\@ngneat\transloco\fesm2022\ngneat-transloco.mjs depends on 'flat'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Is this anything to be concerned about?

We are using Angular 16

**Comment by shaharkazaz** [comment, marked as answer]
@andyb70 That's a [known warning](https://github.com/ngneat/transloco/issues/570) because `flat` is a CommonJS dependency, I'm waiting for v6 to be released which is ESM.

If they don't release it soon I'll consider making my own implementation since Transloco's use case is fairly simple, just nested objects and strings.

I don't think this should prevent migration but that's your call of course 🙂

**Reply by ghost** [reply]
Thank you @shaharkazaz I assumed as much, I've added **flat** to my _allowedCommonJsDependencies_ array for now and I'll await the v6 release.
